docs/design/architecture.md

```diff
@@ -2,128 +2,617 @@
-will be used. `kata-agent` uses this mount point as the root filesystem for the
-container processes.
+## Root filesystem
 
-## Virtualization
+This document uses the term _rootfs_ to refer to a root filesystem
+which is mounted as the top-level directory ("`/`") and often referred
+to as "slash".
+
+It is important to understand this term since the overall system uses
+multiple different rootfs's.
+
+## Example command
+
+The following containerd command creates a container. It is referred
+to throughout this document to help explain various points:
+
+```bash
+$ sudo ctr run --runtime "io.containerd.kata.v2" --rm -t "quay.io/libpod/ubuntu:latest" foo sh
+```
+
+This command requests that containerd:
+
+- Create a container (`ctr run`).
+- Use the Kata [shimv2](#shim-v2-architecture) runtime (`--runtime "io.containerd.kata.v2"`).
+- Delete the container when it [exits](#workload-exit) (`--rm`).
+- Attach the container to the user's terminal (`-t`).
+- Use the Ubuntu Linux [container image](#container-image)
+  to create the container [rootfs](#root-filesystem) that will become
+  the [container environment](#environments)
+  (`quay.io/libpod/ubuntu:latest`).
+- Create the container with the name "`foo`".
+- Run the `sh(1)` command in the Ubuntu rootfs based container
+  environment.
+
+  The command specified here is referred to as the [workload](#workload).
+
+> **Note:**
+>
+> For the purposes of this document and to keep explanations
+> simpler, we assume the user is running this command in the
+> [host environment](#environments).
+
+## Container image
+
+In the [example command](#example-command) the user has specified the
+type of container they wish to run via the container image name:
+`ubuntu`. This image name corresponds to a "container image" that can
+be used to create a container with an Ubuntu Linux environment. Hence,
+in our [example](#example-command), the `sh(1)` command will be run
+inside a container which has an Ubuntu rootfs.
+
+> **Note:**
+>
+> The term "container image" is confusing since the image in question
+> is **not** a container: it is simply a set of files ("an image")
+> that can be used to _create_ a container. The term "container
+> template" would be more accurate but the term "container image" is
+> commonly used so this document uses the standard term.
+
+For the purposes of this document, the most important part of the
+command above is the container image the user has requested. Normally,
+a container image will be "pulled" (downloaded) from a remote site and
+a copy stored locally. This local container image is used by the
+container manager to create an [OCI bundle](#oci-bundle) which will
+form the environment the container will run in.
+
+containerd itself does not create containers, it simply manages them.
+However, before it launches a [runtime](#runtime) to create the
+container, containerd will create an [OCI rootfs bundle](https://github.com/opencontainers/runtime-spec/blob/main/bundle.md).
```

Comment:
Right above this you mention "OCI bundle" and here you mention "OCI rootfs bundle"  Are these two different things?  It seems like they are the same but named slightly differently.